### PR TITLE
#19093 Add Plugin for SalesOrderItemRepository gift message

### PR DESCRIPTION
**Pull request by larsroettig**
### Fixed Issues (if relevant)

1. magento/magento2#19093: API: salesOrderItemRepository does not include gift message

### Manual testing scenarios (*)
1. ...
2. ...

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @larsroettig. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/GiftMessage/Model/Plugin/OrderItemGet.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ *
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\GiftMessage\Model\Plugin;
```
Comment (orlangur) [inline app/code/Magento/GiftMessage/Model/Plugin/OrderItemGet.php:8]: Please move it to `...\GiftMessage\Plugin\...`

File: app/code/Magento/GiftMessage/Model/Plugin/OrderItemGet.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                $orderItem->getItemId()
+            );
+        } catch (NoSuchEntityException $e) {
+            return $orderItem;
+        }
+
+        /** @var \Magento\Sales\Api\Data\OrderItemExtension $orderItemExtension */
+        $orderItemExtension = $extensionAttributes
+            ? $extensionAttributes
+            : $this->orderItemExtensionFactory->create();
```
Comment (orlangur) [inline app/code/Magento/GiftMessage/Model/Plugin/OrderItemGet.php:63]: Simplify it with `?:`.

**Review by orlangur — CHANGES_REQUESTED** [review]
@larsroettig nice!

Please cover new implementation with automated test, I would prefer `api-functional` one.

**Comment by larsroettig** [comment]
@orlangur Jep i will cover this tomorrow with a Test and Fix the Review comments  - thanks for the fast review

File: app/code/Magento/GiftMessage/Model/Plugin/OrderItemGet.php (hunk 1 above)
Comment (larsroettig) [reply app/code/Magento/GiftMessage/Model/Plugin/OrderItemGet.php:8]: All Plugins from this Module is currently placed in` Model\Plugin`

File: app/code/Magento/GiftMessage/Model/Plugin/OrderItemGet.php (hunk 2 above)
Comment (larsroettig) [reply app/code/Magento/GiftMessage/Model/Plugin/OrderItemGet.php:63]: yep this is true i will fix this

**Pull request closed without merging**